src/components/checkbox/GroupableCheckboxReducers.ts

```diff
@@ -0,0 +1,171 @@
+  };
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      parentId: action.payload.id,
+      parent: newCheckbox,
+    };
+  }
+
+  state.parent = newCheckbox;
+  return state;
+};
+
+export const addChildCheckbox = (state: IGroupableCheckboxesState,
+                                 action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  const newChecboxes: ICheckboxState[] = state.checkboxes.concat({
+    id: action.payload.id,
+    checked: !!action.payload.checked,
+  });
+
+  const total: number = state.total + 1;
+  const nbSelected: number = state.nbSelected + (action.payload.checked ? 1 : 0);
+
+  if (state === groupableCheckboxInitialState) {
+    return {
+      ...state,
+      total,
+      nbSelected,
+      checkboxes: newChecboxes,
+      parentId: action.payload.parentId,
+    };
+  }
+
+  return {...state,
+    checkboxes: newChecboxes,
+    total,
+    nbSelected,
+  };
+};
+
+export const groupableCheckboxReducer = (state: IGroupableCheckboxesState = groupableCheckboxInitialState,
+                                         action: IReduxAction<IGroupableCheckboxActionPayload>): IGroupableCheckboxesState => {
+  switch (action.type) {
+    case GroupableCheckboxActions.addGroup:
+      if (action.payload.isParent) {
+        return addParentCheckbox(state, action);
+      }
+      return addChildCheckbox(state, action);
+    case GroupableCheckboxActions.removeGroup:
+      if (state.parentId === action.payload.parentId) {
+        let isChecked: boolean = false;
+        const newCheckboxes = _.reject(state.checkboxes, (checkbox: ICheckboxState) => {
+          const reject: boolean = checkbox.id === action.payload.id;
+          isChecked = reject ? checkbox.checked : isChecked;
+          return reject;
+      });
+
+        const total: number = state.total - 1;
+        const nbSelected: number = isChecked ? state.nbSelected - 1 : state.nbSelected;
+        return {
+          ...state,
+          checkboxes: newCheckboxes,
+          nbSelected,
+          total,
+        };
+      }
+    case GroupableCheckboxActions.toggleGroup:
+      if (state.parentId === action.payload.id || state.parentId === action.payload.parentId) {
+        let isChecked: boolean;
+        if (action.payload.isParent) {
+          isChecked = !state.parent.checked;
+          return {
+            ...state,
+            parent: {
+              id: action.payload.id,
+              checked: isChecked,
+            },
+            checkboxes: _.map(state.checkboxes, (checkbox: ICheckboxState) => {
+              checkbox.checked = isChecked;
+              return checkbox;
+            }),
+            nbSelected: isChecked ? state.total : 0,
+          };
+        }
+        const checkboxes: ICheckboxState[] = _.map(state.checkboxes, (checkbox: ICheckboxState) => {
+          if (checkbox.id === action.payload.id) {
+            isChecked = !checkbox.checked;
```

Comment:
You don't need to assign here, you are returning right after that. Just use `!checkbox.checked` below for the checked property you are returning.

Reply:
This could also simply be a ternary on the return.